README.md

```diff
@@ -518,6 +517,30 @@ channel
 );
 ```
 
+### Symmetric Encryption
+
+Ably client libraries support built-in symmetric encryption of message content, making it easier to build apps that encrypt content fully end-to-end. Whilst TLS is enabled by default and ensures that data is securely sent to and received from Ably, messages are not encrypted within the Ably system. Using the encryption feature of our client libraries ensures that message payloads are opaque, can never be decrypted by Ably, and can only be decrypted by other clients that share your secret key.
+
+1. Get or create a key, typically in one of these ways:
+    - Generating keys on your server and distributing them (i.e. using your own APIs)
+    - Distribute a password and nonce from your server and use a key derivation function (KDF) on clients. An example of a KDF is [HKDF](https://pub.dev/documentation/cryptography/latest/cryptography/Hkdf-class.html) and [PBKDF2](https://pub.dev/documentation/cryptography/latest/cryptography/Pbkdf2-class.html), available in [`package:cryptography`](https://pub.dev/packages/cryptography)
+    - creating a random one using: `ably.Crypto.generateRandomKey();`. You must to distribute this specific key to other clients, since calling `generateRandomKey` on another device will generate a different key.
```

Comment:
I don't think it's good to give different options for generating a key without explaining how the developer should choose between those options, but I also don't think it's useful to try and explain that here, it feels more like something that should be explained in depth in our main documentation and just linked to from individual SDK READMEs. I also don't think it's worth mentioning KDFs here either.

So my suggestion is to have a paragraph a bit like [`ably-ruby`](https://github.com/ably/ably-ruby#symmetric-end-to-end-encrypted-payloads-on-a-channel) and [`ably-js`](https://github.com/ably/ably-js#symmetrical-end-to-end-encrypted-payloads-on-a-channel), followed by an example of generating a key using `ably.Crypto.generateRandomKey()`:

> When a 128 bit or 256 bit key is provided to the library, the `data` attribute of all messages is encrypted and decrypted automatically using that key. The secret key is never transmitted to Ably. See https://www.ably.com/documentation/realtime/encryption.
>
> 1. Create a key by calling `ably.Crypto.generateRandomKey()` (or retrieve one from your server using your own secure API)

Reply:
Okay that makes sense, I've done that in bba9f4c10efd41429285ca0e111a81532eebfb6c

Reply:
Nice, thanks 👍